src/main/kotlin/biokotlin/kmer/Kmer.kt

```diff
@@ -0,0 +1,143 @@
+package biokotlin.kmer
+
+import biokotlin.seq.NucSeq
+import biokotlin.seq.Seq
+
+/**
+ * Create a Kmer from a string
+ * Sequence may be fewer than 32 bases long, left will be padded with A's to equal 32 bases
+ * @param seq String representation of a nucleotide sequence
+ */
+fun Kmer(seq: String): Kmer {
+    if (seq.length > 32) {throw java.lang.IllegalArgumentException("Kmer must be less than or equal to 32 bases long")}
+
+    if( seq.contains(Regex("[^ACTGU]"))) {throw java.lang.IllegalArgumentException("Kmer may contain only A, C, T, G, U")}
+    var encoding = 0L
+    seq.forEach{ c -> encoding = (encoding shl 2) or (c.code.toLong() ushr 1 and 3L) }
+    return Kmer(encoding)
+}
+
+/**
+ * Create a Kmer from a NucSeq
+ * Sequence may be fewer than 32 bases long, left will be padded with A's to equal 32 bases
+ * @param seq NucSeq without ambiguous bases
+ */
+fun Kmer(seq: NucSeq): Kmer {
+    return Kmer(seq.seq())
+}
+
+/**
+ * 2-bit encoded representation of a short nucleotide sequence (32 bases or fewer)
+ * Encoding: A=0, C=1, T/U=2, G=3
+ * For kmers smaller than 32 bp, leftmost digits are padded with 0
+ * Ambiguous bases (eg. N) are not allowed
+ *
+ *
+ * ``` kotlin
+ * val kmer = Kmer(1432462)
+ * val kmerFromString = Kmer("UGCACUGCCA")
+ * val kmerFromSeq = Kmer(NucSeq("TCTCACCTACA")
+ *
+ * val kmerFromString = Kmer("NTACCGANNTCG") // throw error - ambiguous bases
+ * val kmerFromString = Kmer("TTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT") // throw error - string too long
+ * ```
+ */
+@JvmInline
+value class Kmer(val encoding: Long): Comparable<Kmer> {
```

Comment:
General comment/idea,  would it make more sense to pass in the length of the Kmer when you create the Kmer instead of needing to pass in the length in the toString(), toSeq() and reverseCompliment?

Also would it make sense to have a constructor for a String or a NucSeq passed in and it will do the encoding for you within this class?  

I could see the following to be useful:
`val kmer = Kmer("AAATTCGTA")`

Reply:
If we wanted to do that we couldn't have Kmer be a value class, since value classes are restricted to having one property. We chose this format because value classes don't have the memory overhead associated with objects, which was pretty significant compared to the memory footprint of the primitive Long. We also get to take advantage of any optimization the runtime does for primitives. It makes some methods clunkier, like you point out, but I think the performance gains are well worth that.